src/RefractDataStructure.cc

```diff
@@ -489,10 +592,27 @@ namespace drafter {
             }
         }
 
+        template <typename T>
+        struct MakeSectionInfo_ {
+            const bool hasSourceMap;
+            MakeSectionInfo_(bool hasSourceMap) : hasSourceMap(hasSourceMap) {}
+
+            SectionInfo<T> operator()(const T& v, const snowcrash::SourceMap<T>& sm) {
+                return MakeSectionInfo<T>(v, sm, hasSourceMap);
+            }
+        };
+
         template<typename T>
-        void TransformElementData(T* element, ElementData<T>& data) {
+        void TransformElementData(T* element, ElementData<T>& data, bool hasSourceMap) {
+
+            if (data.values.size() != data.valuesSourceMap.size()) {
+                throw std::logic_error("Internal: count of source maps is not equal to count of elements");
```

Comment:
Please throw `snowcrash::Error` as used in this file at other places. And don't use capital letters and the word `internal`.